web/web.go

```diff
@@ -0,0 +1,1022 @@
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+func renameForeignKey(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	pos := r.FormValue("pos")
+	newName := r.FormValue("name")
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	if table == "" || pos == "" || newName == "" {
+		http.Error(w, fmt.Sprintf("Table name or position or new name is empty"), http.StatusBadRequest)
+	}
+	sp := app.conv.SpSchema[table]
+	position, err := strconv.Atoi(pos)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error converting position to integer"), http.StatusBadRequest)
+		return
+	}
+	if position < 0 || position >= len(sp.Fks) {
+		http.Error(w, fmt.Sprintf("No foreign key found at position %d", position), http.StatusBadRequest)
+		return
+	}
+	if !isUniqueName(newName) {
+		http.Error(w, fmt.Sprintf("New name %s is used before in the schema", newName), http.StatusBadRequest)
+		return
+	}
+	sp.Fks[position].Name = newName
+	app.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+func dropSecondaryIndex(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	pos := r.FormValue("pos")
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	if table == "" || pos == "" {
+		http.Error(w, fmt.Sprintf("Table name or position is empty"), http.StatusBadRequest)
+	}
+	sp := app.conv.SpSchema[table]
+	position, err := strconv.Atoi(pos)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error converting position to integer"), http.StatusBadRequest)
+		return
+	}
+	if position < 0 || position >= len(sp.Indexes) {
+		http.Error(w, fmt.Sprintf("No secondary index found at position %d", position), http.StatusBadRequest)
+		return
+	}
+	sp.Indexes = removeSecondaryIndex(sp.Indexes, position)
+	app.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+// updateSessionFile updates the content of session file with
+// latest app.conv.
+func updateSessionFile() error {
+	filePath := app.sessionFile
+	if filePath == "" {
+		return fmt.Errorf("Session file path is empty")
+	}
+	f, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return err
+	}
+	convJSON, err := json.MarshalIndent(app.conv, "", " ")
+	if err != nil {
+		return err
+	}
+	err = f.Truncate(0)
+	if err != nil {
+		return err
+	}
+	if _, err := f.Write(convJSON); err != nil {
+		return err
+	}
+	return nil
+}
+
+// rollback is used to get previous state of conversion in case
+// some unexpected error occurs during update operations.
+func rollback() error {
```

Comment:
Suggest rewriting this function to accept an error and modify it, for two reasons:
a) it allows us to incorporate the original error if the rollback fails (otherwise we lose the original msg)
b) it simplifies the logic for callers: they just need to call:
      return rollback(err)

```
func rollback(err error) error {
	if app.sessionFile != "" {
		app.conv = internal.MakeConv()
		err2 := conversion.ReadSessionFile(app.conv, app.sessionFile)
		if err2 != nil {
			return fmt.Errorf("Encountered error %w. Rollback failed: %v", err, err2)
		}
	}
	return err
}
```

Reply:
done

Reply:
done